File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/molecules/toast/MolToast.kt
package com.perrystreet.woof.designsystem.atomic.molecules.toast

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import com.perrystreet.woof.designsystem.atomic._tokens.sizing.SizingRoles
import com.perrystreet.woof.designsystem.atomic._tokens.spacing.SpacingRoles
import com.perrystreet.woof.designsystem.atomic.atoms.icon.AtomIcon
import com.perrystreet.woof.designsystem.atomic.atoms.icon.roles.IconColorRole
import com.perrystreet.woof.designsystem.atomic.atoms.text.AtomText
import com.perrystreet.woof.designsystem.atomic.atoms.text.roles.TextFontRole
import com.perrystreet.woof.designsystem.theme.Theme
import com.perrystreet.woof.resources.R

@Composable
fun MolToast(text: String) {
    val shape = RoundedCornerShape(Theme.sizing.radiusL)
    Row(
        modifier = Modifier
            .shadow(elevation = Theme.sizing.radiusS, shape = shape)
            .background(color = Theme.colors.surfaceContainer, shape = shape)
            .padding(
                horizontal = Theme.padding.elementExpanded,
                vertical = Theme.padding.elementRelaxed,
            ),
        horizontalArrangement = Arrangement.spacedBy(SpacingRoles.Component.Cozy.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        AtomIcon(
            iconRes = R.drawable.ic_check,
            iconSize = SizingRoles.Icon.S,
            colorRole = IconColorRole.Primary,
        )
        AtomText(
            text = text,
            textFontRole = TextFontRole.SubheadP2,
            maxLines = 2,
        )
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Molecules/Toast/MolToast.swift
import Resources
import SwiftUI

public struct MolToast: View {
    @Environment(\.theme) private var theme

    private let text: String

    public init(text: String) {
        self.text = text
    }

    public var body: some View {
        HStack(spacing: SpacingRoles.Component.cozy.rawValue) {
            AtomIcon(icon: Asset.Icons.check, iconSize: .s, colorRole: .primary)
            AtomText(text: text, textFontRole: .subheadP2, maxLines: 2)
        }
        .padding(.horizontal, PaddingRoles.Element.expanded.rawValue)
        .padding(.vertical, PaddingRoles.Element.relaxed.rawValue)
        .background(theme.colors.surfaceContainer, in: RoundedRectangle(cornerRadius: theme.radius.l))
        .shadow(color: theme.colors.shadow.opacity(theme.alpha.shimmerLowMedium), radius: theme.radius.s)
    }
}
